homeassistant/components/dreo/fan.py

```diff
@@ -0,0 +1,155 @@
+"""Support for Dreo fans."""
+from __future__ import annotations
+
+import logging
+import math
+from functools import cached_property
+from typing import Any
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage, int_states_in_range
+)
+from hscloud.const import DEVICE_TYPE, FAN_DEVICE
+
+from . import MyConfigEntry
+from .entity import DreoEntity
+from hscloud.hscloudexception import HsCloudException, HsCloudBusinessException
+
+_LOGGER = logging.getLogger(__name__)
+
+async def async_setup_entry(
+        hass: HomeAssistant,
+        config_entry: MyConfigEntry,
+        async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Fan from a config entry."""
+
+    async_add_entities([
+        DreoFanHA(device, config_entry)
+        for device in config_entry.runtime_data.devices
+        if DEVICE_TYPE.get(device.get("model")) == FAN_DEVICE.get("type")
+    ])
+
+
+class DreoFanHA(DreoEntity, FanEntity):
+    """Dreo fan."""
+
+    _attr_supported_features = (FanEntityFeature.PRESET_MODE
+                                | FanEntityFeature.SET_SPEED
+                                | FanEntityFeature.OSCILLATE)
+
+    def __init__(self, device, config_entry) -> None:
+        """Initialize the Dreo fan."""
+        super().__init__(device, config_entry)
```

Comment:
Please check https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations

Settings the `_attr_name` in the base entity might be valid for further (future) sensor entities. But the fan control I consider the main feature of the device a therefore the `_attr_name` should be set to `None` here after the init of the super class.